### PR TITLE
refactor: CM Secret UAT and Code Enhancements

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.wrapper.tsx [hunk 1]
```diff
@@ -30,38 +30,38 @@ export const ConfigMapSecretWrapper = (props: CMSecretWrapperProps) => {
...
-        return <ErrorScreenManager code={initError.code} redirectURL={onErrorRedirectURL} reload={reload} />
+    if (appChartRefErr) {
+        return <ErrorScreenManager code={appChartRefErr.code} redirectURL={onErrorRedirectURL} reload={reload} />
     }
 
     return (
         <ConfigMapSecretFormProvider>
             <ConfigMapSecretContainer
                 key={`${CM_SECRET_COMPONENT_NAME[componentType]}-${name}`}
-                appChartRef={initResult[0]?.result}
```
Comment (vivek-devtron) [inline src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.wrapper.tsx:64]: Is this expected?

File: src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx [hunk 2]
```diff
@@ -82,7 +84,13 @@ const EnvConfig = ({ filteredAppIds, envName }: AppGroupDetailDefaultType) => {
         }
     }, [initDataResults])
 
-    if (loading || !envAppList.length) {
+    useEffect(() => {
```
Comment (vivek-devtron) [inline src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:87]: Why is this useEffect required?

File: src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.wrapper.tsx (hunk 1 above)
Comment (RohitRaj011) [reply src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.wrapper.tsx:64]: yes

File: src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx (hunk 2 above)
Comment (RohitRaj011) [reply src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:87]: If the app is unavailable in the current environment, redirect to the app selection page
added comment in code

Comment (vivek-devtron) [reply src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:90]: Can't we achieve the same without useEffect?

Comment (RohitRaj011) [reply src/components/ApplicationGroup/Details/EnvironmentConfig/EnvConfig.tsx:90]: discussed

**Pull request merged**